requirements.txt

```diff
@@ -12,3 +12,4 @@ cryptography>=1.4
 psutil>=2.0.0
 ydiff>=1.2.0
 python-json-logger>=2.0.2
+importlib_resources>=5.4.0
```

Comment:
Sorry, but I am against introducing a new dependency.
According to the [pyinstaller docs](https://www.pyinstaller.org/en/stable/operating-mode.html#how-the-one-folder-program-works) an existing approach supposed to work because the zipfile is extracted to a directory. After that, there is literally no difference.

That is, I am inclined to think that you are solving a wrong problem and have a feeling that your zip file didn't have had any YAML file at all.  In fact, with the [existing spec](https://github.com/zalando/patroni/blob/f6943a859d007fba2d5328a1244cc22b315c3541/patroni.spec#L19-L22) file pyinstaller==5.7.0 currently fails with the error:
```
Unable to find ".../patroni/postgresql/available_parameters/*.yaml" when adding binary and data files.
```

Once the line being removed the single binary successfully built and works (including Postgres v10).

Reply:
I'm not using pyinstaller to package single binary,my intention is solving issue with .pyz format(introduced by PEP 441 – Improving Python ZIP Application Support).
I think they are different,.pyz is just a single file containing multi scripts,we still need an interpreter to execute it.

I have checked validator yml file is presented in extracted archive,and I believe python load the archive directly without extracting to temp directory,so there is no way to read packaged data file by relative path,you'll have to use pkgutil or pkg_resources or stdlib      importlib.resources to load data file by import system.

packaging in python is annoying, although .pyz is a official executable format,it's not a popular package option.
It's reasonable to reject new dependency, maybe we could wrap this part of code, and only use standard lib importlib when python version >= 3.7 ?